File: src/ruby/test/DictionaryTestCase.rb
require 'quickfix_ruby'
require 'test/unit'

class DictionaryTestCase < Test::Unit::TestCase

	def setup
		@object = Quickfix::Dictionary.new
	end

	def test_setGetString
		@object.setString( "STRINGKEY1", "STRINGVALUE1" )
		@object.setString( "STRINGKEY2", "stringvalue2" )

		assert_equal( "STRINGVALUE1", @object.getString("STRINGKEY1") )
		assert_equal( "stringvalue2", @object.getString("STRINGKEY2") )
		assert_equal( "STRINGVALUE2", @object.getString("STRINGKEY2", true) )

		begin
			@object.getString( "STRINGKEY3" )
			assert( false )
		rescue Quickfix::ConfigError
		end
	end

	def test_setGetInt
		@object.setInt( "LONGKEY1", 12 )
		@object.setInt( "LONGKEY2", 9827362 )
		@object.setString( "BADLONGKEY", "AB12" )

		assert_equal( 12, @object.getInt("LONGKEY1") )
		assert_equal( 9827362, @object.getInt("LONGKEY2") )
		
		begin
			@object.getInt( "LONGKEY3" )
			assert( false )
		rescue Quickfix::ConfigError
		end

		begin
			@object.getInt( "BADLONGKEY" )
			assert( false )
		rescue Quickfix::ConfigError
		end
	end

	def test_setGetDouble
		@object.setDouble( "DOUBLEKEY1", 12.3 )
		@object.setDouble( "DOUBLEKEY2", 9827362.9827362 )
		@object.setString( "BADDOUBLEKEY", "AB12.3" )

		assert_equal( 12.3, @object.getDouble("DOUBLEKEY1") )
		assert_equal( 9827362.9827362, @object.getDouble("DOUBLEKEY2") )
		
		begin
			@object.getDouble( "DOUBLEKEY3" )
			assert( false )
		rescue Quickfix::ConfigError
		end

		begin
			@object.getDouble( "BADDOUBLEKEY" )
			assert( false )
		rescue Quickfix::ConfigError
		end
	end

	def test_setGetDay
		@object.setString( "DAY1", "SU" );
		@object.setString( "DAY2", "MO" );
		@object.setString( "DAY3", "TU" );
		@object.setString( "DAY4", "WE" );
		@object.setString( "DAY5", "TH" );
		@object.setString( "DAY6", "FR" );
		@object.setString( "DAY7", "SA" );

		assert( @object.getDay( "DAY1" ) == 1 );
		assert( @object.getDay( "DAY2" ) == 2 );
		assert( @object.getDay( "DAY3" ) == 3 );
		assert( @object.getDay( "DAY4" ) == 4 );
		assert( @object.getDay( "DAY5" ) == 5 );
		assert( @object.getDay( "DAY6" ) == 6 );
		assert( @object.getDay( "DAY7" ) == 7 );

		@object.setDay( "NEXTDAY1", 1 );
		@object.setDay( "NEXTDAY2", 2 );
		@object.setDay( "NEXTDAY3", 3 );
		@object.setDay( "NEXTDAY4", 4 );
		@object.setDay( "NEXTDAY5", 5 );
		@object.setDay( "NEXTDAY6", 6 );
		@object.setDay( "NEXTDAY7", 7 );

		assert( @object.getDay( "NEXTDAY1" ) == 1 );
		assert( @object.getDay( "NEXTDAY2" ) == 2 );
		assert( @object.getDay( "NEXTDAY3" ) == 3 );
		assert( @object.getDay( "NEXTDAY4" ) == 4 );
		assert( @object.getDay( "NEXTDAY5" ) == 5 );
		assert( @object.getDay( "NEXTDAY6" ) == 6 );
		assert( @object.getDay( "NEXTDAY7" ) == 7 );
	end

	def test_merge
		@object.setString( "FIRSTKEY", "FIRSTVALUE" );
		@object.setString( "THIRDKEY", "FIRST" );
		second = Quickfix::Dictionary.new
		second.setString( "SECONDKEY", "SECONDVALUE" );
		second.setString( "THIRDKEY", "SECOND" );
		@object.merge( second );
		assert( @object.getString( "FIRSTKEY" ) == "FIRSTVALUE" );
		assert( @object.getString( "SECONDKEY" ) == "SECONDVALUE" );
		assert( @object.getString( "THIRDKEY" ) == "FIRST" );
	end
end

File: src/python/test/DictionaryTestCase.py
import unittest

import quickfix


class DictionaryTestCase(unittest.TestCase):

    def setUp(self):
        self.object = quickfix.Dictionary()

    def test_setGetString(self):
        self.object.setString("STRINGKEY1", "STRINGVALUE1")
        self.object.setString("STRINGKEY2", "stringvalue2")

        self.assertEqual("STRINGVALUE1", self.object.getString("STRINGKEY1"))
        self.assertEqual("stringvalue2", self.object.getString("STRINGKEY2"))
        self.assertEqual("STRINGVALUE2", self.object.getString("STRINGKEY2", True))

        self.assertRaises(quickfix.ConfigError, self.object.getString, "STRINGKEY3")

    def test_setGetInt(self):
        self.object.setInt("LONGKEY1", 12)
        self.object.setInt("LONGKEY2", 9827362)
        self.object.setString("BADLONGKEY", "AB12")

        self.assertEqual(12, self.object.getInt("LONGKEY1"))
        self.assertEqual(9827362, self.object.getInt("LONGKEY2"))

        self.assertRaises(quickfix.ConfigError, self.object.getInt, "LONGKEY3")
        self.assertRaises(quickfix.ConfigError, self.object.getInt, "BADLONGKEY")

    def test_setGetDouble(self):
        self.object.setDouble("DOUBLEKEY1", 12.3)
        self.object.setDouble("DOUBLEKEY2", 9827362.9827362)
        self.object.setString("BADDOUBLEKEY", "AB12.3")

        self.assertEqual(12.3, self.object.getDouble("DOUBLEKEY1"))
        self.assertEqual(9827362.9827362, self.object.getDouble("DOUBLEKEY2"))

        self.assertRaises(quickfix.ConfigError, self.object.getDouble, "DOUBLEKEY3")
        self.assertRaises(quickfix.ConfigError, self.object.getDouble, "BADDOUBLEKEY")

    def test_setGetDay(self):
        self.object.setString("DAY1", "SU")
        self.object.setString("DAY2", "MO")
        self.object.setString("DAY3", "TU")
        self.object.setString("DAY4", "WE")
        self.object.setString("DAY5", "TH")
        self.object.setString("DAY6", "FR")
        self.object.setString("DAY7", "SA")

        self.assertEqual(self.object.getDay("DAY1"), 1)
        self.assertEqual(self.object.getDay("DAY2"), 2)
        self.assertEqual(self.object.getDay("DAY3"), 3)
        self.assertEqual(self.object.getDay("DAY4"), 4)
        self.assertEqual(self.object.getDay("DAY5"), 5)
        self.assertEqual(self.object.getDay("DAY6"), 6)
        self.assertEqual(self.object.getDay("DAY7"), 7)

        self.object.setDay("NEXTDAY1", 1)
        self.object.setDay("NEXTDAY2", 2)
        self.object.setDay("NEXTDAY3", 3)
        self.object.setDay("NEXTDAY4", 4)
        self.object.setDay("NEXTDAY5", 5)
        self.object.setDay("NEXTDAY6", 6)
        self.object.setDay("NEXTDAY7", 7)

        self.assertEqual(self.object.getDay("NEXTDAY1"), 1)
        self.assertEqual(self.object.getDay("NEXTDAY2"), 2)
        self.assertEqual(self.object.getDay("NEXTDAY3"), 3)
        self.assertEqual(self.object.getDay("NEXTDAY4"), 4)
        self.assertEqual(self.object.getDay("NEXTDAY5"), 5)
        self.assertEqual(self.object.getDay("NEXTDAY6"), 6)
        self.assertEqual(self.object.getDay("NEXTDAY7"), 7)

    def test_merge(self):
        self.object.setString("FIRSTKEY", "FIRSTVALUE")
        self.object.setString("THIRDKEY", "FIRST")
        second = quickfix.Dictionary()
        second.setString("SECONDKEY", "SECONDVALUE")
        second.setString("THIRDKEY", "SECOND")
        self.object.merge(second)
        self.assertEqual(self.object.getString("FIRSTKEY"), "FIRSTVALUE")
        self.assertEqual(self.object.getString("SECONDKEY"), "SECONDVALUE")
        self.assertEqual(self.object.getString("THIRDKEY"), "FIRST")

if __name__ == '__main__':
    unittest.main(verbosity=2)
